### PR TITLE
Decompose stablehlo transpose into multiple ttir.transpose ops

### PR DESCRIPTION
**Pull request by LPanosTT**
The scope of `stablehlo.transpose` includes general permutation which is equivalent to multiple dimension swaps. These changes to stablehlo -> TTIR conversion allows the decomposition of `stablehlo.transpose` into however many `ttir.transpose` ops are required to perform the desired permutation.

### CODE REVIEW

File: test/ttmlir/Conversion/StableHLOToTTIR/transpose_op.mlir [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// REQUIRES: stablehlo
```
Comment (mmanzoorTT) [inline test/ttmlir/Conversion/StableHLOToTTIR/transpose_op.mlir:1]: Please move this test to `test/ttmlir/Conversion/StableHLOToTTIR/unary/transpose_op.mlir`

Comment (LPanosTT) [reply test/ttmlir/Conversion/StableHLOToTTIR/transpose_op.mlir]: Done

Comment (LPanosTT) [reply test/ttmlir/Conversion/StableHLOToTTIR/transpose_op.mlir]: I also changed the name of the transpose test that was originally there.

**Comment by LPanosTT** [comment]
> TTNN is supporting transpose op for higher dimensions with recent tt-metal uplift. PR #1288 uses `ttnn.transpose` instead of decomposing it to permutation. May be we should follow the same and lower `stablehlo.transpose` to `ttir.transpose`

It looks like they are just making a runtime change to use `ttnn::transpose` in runtime instead of using a single swap permute.

**Review by uazizTT — APPROVED** [review]

**Pull request merged**